### DISCUSSION TITLE
Feature Request: Force Icon and Icon Color

### DISCUSSION THREAD

**Discussion by N0thingSpecial** [cgiesche/streamdeck-homeassistant#243] [Feature Requests]
Hi,

first of all: thanks for the great Plugin! If I could make one wish, it would be to add Custom Icons and Custom Icon Colors. Since you already check for certain device_groups in ["entityConfigFactory.js"](https://github.com/cgiesche/streamdeck-homeassistant/blob/d2e48ad3ea73c486862f5a06406c5e30531dbb1f/src/modules/plugin/entityConfigFactory.js#L210), would it be possible to have an option to overwrite the icon color. Perhabs even with depending on the {{sate}} variable?

I saw there was already one Request #173. Did you decide against this feature?

Thanks in Advance.

**Comment by cgiesche** [comment]
Hi!

I'm currently working on a completely new implementation of button rendering that includes per-button-options for choosing between the icon provided by Home Assistant, the plugin, or no icon at all.

The internal plugin rendering will be powerful and declaratively configurable, allowing for the definition of various icons and colors for different combinations of domains, entities, and states. I will provide a default configuration. However, I'm still considering how to best allow users to configure these settings. They may be not trivial ;)

Example of how the declarative configuration file will look:
```
cover:
  icon: mdi:garage
  color: "#00AA00"
  states:
    open:
      icon: mdi:garage-open
      color: "#AA0000"
    closing:
      icon: mdi:arrow-down-box
    opening:
      icon: mdi:arrow-up-box

sensor:
  classes:
    battery:
      icon: |
        !nunjucks
        {% set numState = state | int %}
        {% if numState <= 10 %}
          mdi:battery-outline
        {% elif numState <= 20 %}
          mdi:battery-10
        {% elif numState <= 30 %}
          mdi:battery-20
        {% elif numState <= 40 %}
          mdi:battery-30
        {% elif numState <= 50 %}
          mdi:battery-40
        {% elif numState <= 60 %}
          mdi:battery-50
        {% elif numState <= 70 %}
          mdi:battery-60
        {% elif numState <= 80 %}
          mdi:battery-70
        {% elif numState <= 90 %}
          mdi:battery-80
        {% else %}
          mdi:battery
        {% endif %}
      color: |
        !nunjucks
        {% set numState = state | int %}
        {% if numState <= 10 %}
          #ff0000
        {% elif numState <= 20 %}
          #e21c00
        {% elif numState <= 30 %}
          #c63800
        {% elif numState <= 40 %}
          #aa5500
        {% elif numState <= 50 %}
          #8d7100
        {% elif numState <= 60 %}
          #718d00
        {% elif numState <= 70 %}
          #55aa00
        {% elif numState <= 80 %}
          #38c600
        {% elif numState <= 90 %}
          #1ce200
        {% else %}
          #00ff00
        {% endif %}
```

**Reply by jer78** [reply]
Is this available yet or in a future update? Wondering how to get custom icons based on a state. Thanks

**Reply by fkthisusrrequirement** [reply]
Yes, it is available.

Look at the comments below in this link: https://github.com/cgiesche/streamdeck-homeassistant/issues/270
